cluster-autoscaler/simulator/predicatechecker/schedulerbased_test.go

```diff
@@ -102,27 +175,107 @@ func TestFitsAnyNode(t *testing.T) {
 	n1000 := BuildTestNode("n1000", 1000, 2000000)
 	n2000 := BuildTestNode("n2000", 2000, 2000000)
 
-	var err error
+	defaultPredicateChecker, err := NewTestPredicateChecker(nil)
+	assert.NoError(t, err)
 
-	clusterSnapshot := clustersnapshot.NewBasicClusterSnapshot()
-	err = clusterSnapshot.AddNode(n1000)
+	// temp dir
+	tmpDir, err := os.MkdirTemp("", "scheduler-configs")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	customConfigFile := filepath.Join(tmpDir, "custom_config.yaml")
+	if err := os.WriteFile(customConfigFile, []byte(`
+apiVersion: kubescheduler.config.k8s.io/v1
+kind: KubeSchedulerConfiguration
+profiles:
+- pluginConfig:
+  plugins:
+    multiPoint:
+      disabled:
+      - name: NodeResourcesFit
+        weight: 1
+  schedulerName: custom-scheduler`), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	customConfig, err := scheduler.GetSchedulerConfig(customConfigFile)
 	assert.NoError(t, err)
-	err = clusterSnapshot.AddNode(n2000)
+	customPredicateChecker, err := NewTestPredicateChecker(customConfig)
 	assert.NoError(t, err)
 
-	predicateChecker, err := NewTestPredicateChecker()
-	assert.NoError(t, err)
+	testCases := []struct {
+		name             string
+		predicateChecker PredicateChecker
+		pod              *apiv1.Pod
+		expectedNodes    []string
+		expectError      bool
+	}{
+		// default predicate checker test cases
+		{
+			name:             "default - small pod - no error",
+			predicateChecker: defaultPredicateChecker,
+			pod:              p900,
+			expectedNodes:    []string{"n1000", "n2000"},
+			expectError:      false,
+		},
+		{
+			name:             "default - medium pod - no error",
+			predicateChecker: defaultPredicateChecker,
+			pod:              p1900,
+			expectedNodes:    []string{"n2000"},
+			expectError:      false,
+		},
+		{
+			name:             "default - large pod - insufficient cpu",
+			predicateChecker: defaultPredicateChecker,
+			pod:              p2100,
+			expectError:      true,
+		},
 
-	nodeName, err := predicateChecker.FitsAnyNode(clusterSnapshot, p900)
-	assert.NoError(t, err)
-	assert.True(t, nodeName == "n1000" || nodeName == "n2000")
+		// custom predicate checker test cases
+		{
+			name:             "custom - small pod - no error",
```

Comment:
Test cases for custom predicate checker are a copy of the default predicate checker's test cases except we don't expect an error for any of the custom predicate checker test cases.